src/vivarium/framework/population.py

```diff
@@ -211,41 +210,41 @@ def update(self, population_update: Union[pd.DataFrame, pd.Series]):
             type inconsistent with the original population data.
 
         """
-        if population_update.empty:
-            return
-
-        # TODO: Cast series to data frame and clean this up.
         if isinstance(population_update, pd.Series):
-            if population_update.name in self._columns:
-                affected_columns = [population_update.name]
-            elif len(self._columns) == 1:
-                affected_columns = self._columns
-            else:
-                raise PopulationError(
-                    "Cannot update with a pandas series unless the series name is a column "
-                    "name in the view or there is only a single column in the view."
-                )
-        else:
-            if not set(population_update.columns).issubset(self._columns):
-                raise PopulationError(
-                    f"Cannot update with a DataFrame that contains columns the view does not. "
-                    f"Dataframe contains the following extra columns: "
-                    f"{set(population_update.columns).difference(self._columns)}."
-                )
-            affected_columns = set(population_update.columns)
+            if population_update.name is None:
+                if len(self.columns) == 1:
+                    population_update.name = self.columns[0]
+                else:
+                    raise PopulationError(
+                        "Cannot update with an unnamed pandas series unless there is only a single "
+                        "column in the view."
+                    )
+
+            population_update = pd.DataFrame(population_update)
+
+        affected_columns = set(population_update.columns)
+
+        if not affected_columns.issubset(self.columns):
+            raise PopulationError(
+                f"Cannot update with a DataFrame or Series that contains columns the view does "
+                f"not. Dataframe contains the following extra columns: "
+                f"{set(population_update.columns).difference(self.columns)}."
+            )
+
+        if population_update.empty and not affected_columns.difference(self._manager.columns):
```

Comment:
I'm not sure why we would be passing in an empty population update/if that would have any effect on the population. As a result I'm also not sure why this boolean is making the second check. I read it as checking that there are no columns in population_update that aren't in the state table, but why is this relevant if population_update is empty?